src/domain/knight/KnightLogic.ts

```diff
@@ -1,60 +1,41 @@
 import { Result } from 'coderitter-api-remote-method-call'
-import { ReadCriteria, Criteria } from 'knight-criteria'
+import { Criteria } from 'knight-criteria'
 import { Log } from 'knight-log'
-import { create, delete_, read, update, count } from 'knight-orm'
+import { Orm } from 'knight-orm'
 import Knight from './Knight'
 import { PgTransaction } from 'knight-pg-transaction'
-import { EntityResult, EntitiesResult, CreateOrGetResult, CountResult } from '../api'
-import schema from '../DbSchema'
+import { EntityResult, EntitiesResult, CountResult } from '../api'
 import { txQuery } from '../../util/transactions'
-import { KnightCreateValidator, KnightDeleteValidator, KnightUpdateValidator } from './validators'
+import { KnightStoreValidator, KnightDeleteValidator } from './validators'
 
 let log = new Log('KnightLogic.ts')
 
 export default class KnightLogic {
 
-    async create(knight: Partial<Knight>, tx: PgTransaction): Promise<CreateOrGetResult<Knight>> {
-        let l = log.mt('create')
+    orm: Orm
 
-        return tx.runInTransaction(async () => {
-            let validator = new KnightCreateValidator(tx)
-            let misfits = await validator.validate(knight)
-            l.dev('Validation yields the following misfits', misfits)
-
-            if (misfits.length > 0) {
-                await tx.rollback()
-                return Result.misfits(misfits) as any
-            }
-
-            let created = await create(schema, 'knight', 'mysql', txQuery(tx), knight) as Knight //not working with mariaDB?
-
-            l.dev('created', created)
-
-            let knightCreateResult = new CreateOrGetResult(created)
-            l.dev('knightCreateResult', knightCreateResult.entity)
-      
-            return knightCreateResult
-        })
+    constructor(orm:Orm){
```

Comment:
Mach die Abhängigkeit zu dem ORM nicht in den Constructor, auch wenn das semantisch auf jeden Fall die richtige Idee wäre. Der Hintergrund ist, dass die Abhängigkeiten so kreuz und quer werden können, dass es einen Deadlock zwischen den Konstruktoren geben kann. Ist zwar hier konkret nicht so, aber damit man nicht zwei Patterns hat, machen wir es lieber so, dass die Abhängigkeit ein Property der Klasse mit Ausrufezeichen wird: orm!: Orm. Das Ausrufezeichen, damit man es nicht im Constructor initialisieren muss und damit man innerhalb der Klasse dann nicht jedes Mal prüfen muss, ob der ORM gesetzt ist.